connectivity/FEATURE_BLE/source/generic/GapImpl.cpp

```diff
@@ -2239,43 +2178,274 @@ void Gap::on_extended_advertising_report(
     const uint8_t *data
 )
 {
-#if BLE_FEATURE_PRIVACY
-    // Check if the address hasn't been resolved
-    if (_privacy_enabled &&
-        _central_privacy_configuration.resolution_strategy == central_privacy_configuration_t::RESOLVE_AND_FILTER &&
-        address_type != nullptr &&
-        *address_type == connection_peer_address_type_t::RANDOM_ADDRESS &&
-        is_random_private_resolvable_address(address.data())
-        ) {
-        // FIXME: Implement
+    if (!_event_handler) {
         return;
     }
+
+    peer_address_type_t peer_address_type = address_type ? (peer_address_type_t::type) address_type->value()
+                                                         : peer_address_type_t::ANONYMOUS;
+
+    AdvertisingReportEvent event(
+        BLE_ERROR_NONE,
+        event_type,
+        peer_address_type,
+        address,
+        primary_phy,
+        secondary_phy ? *secondary_phy : phy_t::NONE,
+        advertising_sid,
+        tx_power,
+        rssi,
+        periodic_advertising_interval,
+        (ble::peer_address_type_t::type) direct_address_type.value(),
+        (address_t &) direct_address,
+        make_Span(data, data_length)
+    );
+
+    signal_advertising_report(
+        event
+    );
+}
+
+#if BLE_FEATURE_CONNECTABLE
+void Gap::signal_connection_complete(
+    ConnectionCompleteEvent& event
+)
+{
+#if BLE_FEATURE_PRIVACY
+    bool address_resolved = false;
+
+    /* if address resolution is not needed or already handled then the address is already resolved */
+    if (!_privacy_enabled ||
+        _address_registry.is_controller_privacy_supported() ||
+        !is_random_private_resolvable_address(event.getPeerAddress())) {
```

Comment:
The advertising type should be checked, public addresses may be formatted like resolvable addresses.